### PR TITLE
[#KS-17492] Backend api sync 

### PR DESCRIPTION
**Pull request by SAkshat**
https://kitchenstadium.atlassian.net/browse/KS-17492

### CODE REVIEW

File: app/models/spree/affiliate.rb [hunk 1]
```diff
@@ -60,7 +59,8 @@ def create_user
         @user = Spree::User.find_or_initialize_by(email: email)
         self.active_on_create = true if user.persisted?
         affiliate_role = Spree::Role.affiliate
-        user.spree_roles << affiliate_role unless user.spree_roles.include?(affiliate_role)
+        user.save!
```
Comment (siddharth28) [inline app/models/spree/affiliate.rb]: @SAkshat why do we need this?

File: app/views/spree/admin/affiliates/transactions.html.erb [hunk 2]
```diff
@@ -35,20 +35,23 @@
...
         </th>
-        <th><%= sort_link @search, :amount, Spree.t(:amount), { default_order: "desc" }, {title: 'admin_products_listing_amount'} %></th>
+        <th><%= Spree.t(:order_total) %></th>
+        <th><%= sort_link @search, :amount, Spree.t(:commission_amount), { default_order: "desc" }, {title: 'admin_products_listing_amount'} %></th>
         <th><%= Spree.t(:locked) %></th>
       </tr>
     </thead>
     <tbody>
       <% @commission_transactions.each do |transaction|%>
+        <% order = Spree::Order.find_by(number: transaction.commissionable.number) %>
```
Comment (siddharth28) [inline app/views/spree/admin/affiliates/transactions.html.erb]: @SAkshat can we avoid n+1 queries here?

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Spree::Core::Engine.routes.draw do
   get 'r/:code' => 'reffiliate#referral', as: 'referral'
-  get 'a/:path' => 'reffiliate#affiliate', as: 'affiliate'
+  get 'partner/:path' => 'reffiliate#affiliate', as: 'affiliate'
```
Comment (siddharth28) [inline config/routes.rb:3]: @SAkshat i don't think we are using these routes.

Comment (SAkshat) [reply config/routes.rb:3]: @siddharth28 - Removed unused routes. Have kept `/partner/:path` as we are using url helpers in the code.

**Review by siddharth28 — APPROVED** [review]

**Pull request merged**